packages/front-end/components/Segments/SegmentForm.tsx

```diff
@@ -60,6 +69,22 @@ const SegmentForm: FC<{
 
   const datasource = getDatasourceById(form.watch("datasource"));
 
+  const filteredProjects = projects.filter((project) => {
+    // only filter projects is the data source isn't in All Projects (aka, projects is an empty array)
+    if (datasource?.projects && datasource.projects.length) {
+      return (
+        datasource.projects.includes(project.id) ||
+        form.watch("projects").includes(project.id)
+      );
+    }
+  });
+
+  const projectOptions = useProjectOptions(
+    (project) => permissionsUtil.canCreateMetric({ projects: [project] }), //TODO: Do we need to check here for the create vs update permission?
```

Comment:
Did you get an answer for this question? From my reading of things it's correct to use canCreate but maybe you found a reason that isn't right